sql/dj_setup_database.in

```diff
@@ -354,6 +370,64 @@ upgrade)
 	verbose "DOMjudge database upgrade completed."
 	;;
 
+snapshot-dump)
+	read_dbpasswords
+	if [ -z "$DUMPNAME" ]; then
+		usage
+		exit 1
+	fi
+
+	if [ -f "${DATABASEDUMPDIR}/${DUMPNAME}.sql.gz" ]; then
+		while true; do
+			read -p "Overwrite existing database dump (y/N)? " yn
+			case $yn in
+				[Yy]* ) create_database_dump $DUMPNAME ;;
+				''|[Nn]* ) break;;
+			esac
+		done
+	else
+		create_database_dump $DUMPNAME
+	fi
+
+	read -p "Provide prefix or say 'no' to abort. Format: [prefix]-$1 " prefix
+	case $prefix in
+		''|[Nn]o|[Nn] ) exit 2; break;;
+		* ) create_database_dump ${prefix}-${DUMPNAME}; break;;
+	esac
+	;;
+
+snapshot-load)
```

Comment:
Most often you are loading a new enough or compatible enough database, but we might want to consider calling here `uninstall` and `create-db-users` first. Then we are in a known "good" state.

Reply:
We can't do this IMO, the `domjudge` user from dbpasswords doesn't have the privilige to drop it's own user.

I assume we don't want to dump as the root user all the time?

Reply:
Load should now be done as `root`.